tests/Feature/Connect/Actions/BuildClientPatchDataActionTest.php

```diff
@@ -421,7 +424,267 @@ public function testLocallyOptedOutOfSubsetsAndLoadedOptedOutSubsetHash(): void
+
+        // Act
+        $result = (new BuildClientPatchDataAction())->execute(gameHash: $specialtyGameHash, user: $user);
+
+        // Assert
+        $this->assertTrue($result['Success']);
+        $this->assertEquals($baseGame->RichPresencePatch, $result['PatchData']['RichPresencePatch']);
+    }
+
+    public function testItUsesCoreGameRichPresenceForBonusSet(): void
+    {
+        // Arrange
+        $baseGame = $this->createGameWithAchievements(
+            $this->system,
+            'Dragon Quest III',
+            publishedCount: 1
+        );
+        $bonusGame = $this->createGameWithAchievements(
+            $this->system,
+            'Dragon Quest III [Subset - Bonus]',
+            publishedCount: 2
+        );
+
+        $this->upsertGameCoreSetAction->execute($baseGame);
+        $this->upsertGameCoreSetAction->execute($bonusGame);
+        $this->associateAchievementSetToGameAction->execute($baseGame, $bonusGame, AchievementSetType::Bonus, 'Bonus');
+
+        $bonusGameHash = GameHash::factory()->create(['game_id' => $bonusGame->id]);
+        $user = User::factory()->create(['websitePrefs' => self::OPT_IN_TO_ALL_SUBSETS_PREF_ENABLED]);
+
+        // Act
+        $result = (new BuildClientPatchDataAction())->execute(gameHash: $bonusGameHash, user: $user);
+
+        // Assert
+        $this->assertTrue($result['Success']);
+        $this->assertEquals($baseGame->RichPresencePatch, $result['PatchData']['RichPresencePatch']);
+    }
+
+    public function testItResolvesRootDataCorrectlyForSpecialtySet(): void
+    {
+        // Arrange
+        $baseGame = $this->createGameWithAchievements(
+            $this->system,
+            'Dragon Quest III',
+            publishedCount: 3,
+        );
+        $specialtyGame = $this->createGameWithAchievements(
+            $this->system,
+            'Dragon Quest III [Subset - Special]',
+            publishedCount: 2,
+        );
+
+        $this->upsertGameCoreSetAction->execute($baseGame);
+        $this->upsertGameCoreSetAction->execute($specialtyGame);
+        $this->associateAchievementSetToGameAction->execute($baseGame, $specialtyGame, AchievementSetType::Specialty, 'Special');
+
+        $specialtyGameHash = GameHash::factory()->create(['game_id' => $specialtyGame->id]);
+        $user = User::factory()->create(['websitePrefs' => self::OPT_IN_TO_ALL_SUBSETS_PREF_ENABLED]);
+
+        // Act
+        $result = (new BuildClientPatchDataAction())->execute(gameHash: $specialtyGameHash, user: $user);
+
+        // Assert
+        $this->assertTrue($result['Success']);
+
+        // ... root data should be from the base game ...
+        $this->assertEquals($baseGame->id, $result['PatchData']['ID']);
+        $this->assertEquals($baseGame->title, $result['PatchData']['Title']);
+        $this->assertEquals($baseGame->ImageIcon, $result['PatchData']['ImageIcon']);
+        $this->assertCount(3, $result['PatchData']['Achievements']); // the base game's achievements
```

Comment:
I'm not convinced this is right. If I'm playing a specialty set, that's what I'm focused on and the core achievements are "bonus".